pkg/client/cloudevents/grpcsource/watcherstore.go

```diff
@@ -0,0 +1,160 @@
+package grpcsource
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/openshift-online/maestro/pkg/api/openapi"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+
+	workv1 "open-cluster-management.io/api/work/v1"
+
+	"open-cluster-management.io/sdk-go/pkg/cloudevents/generic/types"
+	"open-cluster-management.io/sdk-go/pkg/cloudevents/work/common"
+	"open-cluster-management.io/sdk-go/pkg/cloudevents/work/store"
+	"open-cluster-management.io/sdk-go/pkg/cloudevents/work/utils"
+)
+
+// RESTFullAPIWatcherStore implements the WorkClientWatcherStore interface, it is
+// used to build a source work client. The work client uses this store to
+//   - get/list works from Maestro server via RESTfull APIs
+//   - receive the work status update and send the updated work to the watch channel
+type RESTFullAPIWatcherStore struct {
```

Comment:
s/RESTFullAPIWatcherStore/RESTFulAPIWatcherStore
You have extra `l`

Reply:
corrected